src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    public class EventLogTests
+    {
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void EventLogReinitializationException()
+        {
+            using (EventLog eventLog = new EventLog())
+            {
+                eventLog.BeginInit();
+                Assert.Throws<InvalidOperationException>(() => eventLog.BeginInit());
+                eventLog.Close();
+            }
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void ClearLogTest()
+        {
+            if (!AdminHelpers.IsProcessElevated())
+                return;
+
+            string source = Guid.NewGuid().ToString("N");
+            EventLog.CreateEventSource(source, "MyNewLog");
```

Comment:
This should be deleted (in a finally) with `EventLog.DeleteEventSource`.

Also does the "MyNewLog" need to be a random string?